syntax = "proto3";

import "google/protobuf/timestamp.proto";

message SInt64Value
{
    sint64 value = 1;
}


enum OrderStatus {
    CREATED = 0;
    CANCELED = 1;
    PAID = 2;
    FINISHED = 3;
}

message OrderInput {
    sint64 id = 1;
    sint64 account_id = 2;
    map<string, sint64> items = 4;
    google.protobuf.Timestamp create_time = 5;
    string memo = 6;
}

message Order {
    sint64 id = 1;
    sint64 account_id = 2;
    OrderStatus status = 3;
    map<string, sint64> items = 4;
    google.protobuf.Timestamp create_time = 5;
    string memo = 6;
}

message FetchOrdersInput {
    sint64 account_id = 1;
    sint64 start_order_id = 2;
    sint64 limit = 3;
}

message Orders {
    repeated Order value = 1;
}